### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [PixarAnimationStudios/USD#9]
Would you like to add more error handling for return values from functions like the following?
- [fprintf](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fprintf.html "Print formatted output.") ⇒ [TfScriptModuleLoader::WriteDotFile](https://github.com/PixarAnimationStudios/USD/blob/2eb01f5cd4c2dae4e1ef9912ca27a93083bb6ef4/pxr/base/lib/tf/scriptModuleLoader.cpp#L163)
- [sigaction](http://pubs.opengroup.org/onlinepubs/9699919799/functions/realloc.html "Change a signal action.") ⇒ [TfInstallTerminateAndCrashHandlers](https://github.com/PixarAnimationStudios/USD/blob/2eb01f5cd4c2dae4e1ef9912ca27a93083bb6ef4/pxr/base/lib/tf/diagnostic.cpp#L210)

**Comment by amooty** [comment]
Filed as internal issue #135161.

**Comment by hikaru-light** [comment]
I agree. So I will also check responses from some functions as possible.
